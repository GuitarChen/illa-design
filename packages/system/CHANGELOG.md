# @illa-design/system

## 0.0.10

### Patch Changes

- fix bugs

## 0.0.9

### Patch Changes

- fix bugs

## 0.0.8

### Patch Changes

- fix bugs

## 0.0.7

### Patch Changes

- bugfix

## 0.0.6

### Patch Changes

- fix bugs

## 0.0.5

### Patch Changes

- update version

## 0.0.4

### Patch Changes

- update version

## 0.0.3

### Patch Changes

- fix some bugs

## 0.0.2

### Patch Changes

- update all components version

## 0.0.1

### Patch Changes

- release the first version

## 0.0.0

### Patch Changes

- init publish
