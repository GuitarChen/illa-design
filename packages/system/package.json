{
  "name": "@illa-design/system",
  "version": "1.0.16",
  "description": "Illa Design base.",
  "keywords": [
    "system",
    "react",
    "component"
  ],
  "sideEffects": false,
  "author": "ILLA Cloud <opensource@illasoft.com>",
  "homepage": "https://github.com/illacloud/illa-design",
  "license": "Apache-2.0",
  "main": "dist/umd/index.js",
  "module": "dist/es/index.js",
  "types": "dist/types/index.d.ts",
  "typings": "dist/types/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/es/index.js",
      "require": "./dist/umd/index.js"
    }
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/illacloud/illa-design.git",
    "directory": "packages/system"
  },
  "bugs": {
    "url": "https://github.com/illacloud/illa-design/issues"
  },
  "dependencies": {
    "dayjs": "^1.11.7"
  },
  "scripts": {
    "build": "vite build"
  },
  "devDependencies": {
    "@vitejs/plugin-react-swc": "^3.0.1",
    "@rollup/plugin-typescript": "^10.0.1"
  }
}
