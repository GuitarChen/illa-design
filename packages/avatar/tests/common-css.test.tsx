import {
  applyAvatarContainer,
  applyAvatarShapeCircle,
  applyAvatarShapeSquare,
  applyAvatarSize,
  applyBgColor,
  applyMergeCss,
  applyShape,
} from "../src/style"
import { BsFacebook } from "react-icons/bs"
import { omit } from "@illa-design/system"

test("Common css renders background color", () => {
  expect(omit(applyBgColor("green"), ["name"])).toMatchSnapshot()
  expect(omit(applyBgColor("gray"), ["name"])).toMatchSnapshot()
  expect(omit(applyBgColor("#123321"), ["name"])).toMatchSnapshot()
})

test("Common css renders square shape", () => {
  expect(omit(applyAvatarShapeSquare("small"), ["name"])).toMatchSnapshot()
  expect(omit(applyAvatarShapeSquare("medium"), ["name"])).toMatchSnapshot()
  expect(omit(applyAvatarShapeSquare("large"), ["name"])).toMatchSnapshot()
})

test("Common css renders circle shape", () => {
  expect(omit(applyAvatarShapeCircle("small"), ["name"])).toMatchSnapshot()
  expect(omit(applyAvatarShapeCircle("medium"), ["name"])).toMatchSnapshot()
  expect(omit(applyAvatarShapeCircle("large"), ["name"])).toMatchSnapshot()
})

test("Common css renders container", () => {
  expect(omit(applyAvatarContainer("green"), ["name"])).toMatchSnapshot()
  expect(omit(applyAvatarContainer("gray"), ["name"])).toMatchSnapshot()
  expect(omit(applyAvatarContainer("#123321"), ["name"])).toMatchSnapshot()
})

test("Common css renders size", () => {
  expect(omit(applyAvatarSize("small"), ["name"])).toMatchSnapshot()
  expect(omit(applyAvatarSize("medium"), ["name"])).toMatchSnapshot()
  expect(omit(applyAvatarSize("large"), ["name"])).toMatchSnapshot()
})

test("Common css renders shape", () => {
  expect(applyShape("circle")).toBe("50%")
  expect(applyShape("square")).toBe("4px")
})

test("Common css renders merge", () => {
  expect(
    omit(
      applyMergeCss({
        colorScheme: "gray",
        size: "large",
        shape: "square",
        icon: <BsFacebook />,
      }),
      ["name"],
    ),
  ).toMatchSnapshot()

  expect(
    omit(
      applyMergeCss({
        colorScheme: "gray",
        size: "large",
        shape: "square",
        src: "https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png",
      }),
      ["name"],
    ),
  ).toMatchSnapshot()

  expect(
    omit(
      applyMergeCss({
        colorScheme: "gray",
        size: "large",
        shape: "square",
        text: "Hello World",
      }),
      ["name"],
    ),
  ).toMatchSnapshot()
})
