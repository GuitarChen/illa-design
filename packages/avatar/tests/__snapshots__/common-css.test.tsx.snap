// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common css renders background color 1`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
        background-color: #007a41;
      ",
  "toString": [Function],
}
`;

exports[`Common css renders background color 2`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
        background-color: #c2c2c2;
      ",
  "toString": [Function],
}
`;

exports[`Common css renders background color 3`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
      background-color: #123321;
    ",
  "toString": [Function],
}
`;

exports[`Common css renders circle shape 1`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    width: 32px;
    height: 32px;
    font-size: 14px;
    line-height: 22px;
    border-radius: 50%;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders circle shape 2`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    width: 40px;
    height: 40px;
    font-size: 14px;
    line-height: 22px;
    border-radius: 50%;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders circle shape 3`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    width: 64px;
    height: 64px;
    font-size: 24px;
    line-height: 32px;
    border-radius: 50%;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders container 1`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    
        background-color: #007a41;
      ;;
    display: inline-flex;
    overflow: hidden;
    word-wrap: break-word;
    white-space: nowrap;
    color: #fff;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding: 4px;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders container 2`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    
        background-color: #c2c2c2;
      ;;
    display: inline-flex;
    overflow: hidden;
    word-wrap: break-word;
    white-space: nowrap;
    color: #fff;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding: 4px;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders container 3`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    
      background-color: #123321;
    ;;
    display: inline-flex;
    overflow: hidden;
    word-wrap: break-word;
    white-space: nowrap;
    color: #fff;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding: 4px;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders merge 1`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    
    
        background-color: #c2c2c2;
      ;;
    display: inline-flex;
    overflow: hidden;
    word-wrap: break-word;
    white-space: nowrap;
    color: #fff;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding: 4px;
  ;;
    
    width: 64px;
    height: 64px;
    font-size: 24px;
    line-height: 32px;
    border-radius: 4px;
  ;;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders merge 2`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    
    
        background-color: #c2c2c2;
      ;;
    display: inline-flex;
    overflow: hidden;
    word-wrap: break-word;
    white-space: nowrap;
    color: #fff;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding: 4px;
  ;;
    
    width: 64px;
    height: 64px;
    font-size: 24px;
    line-height: 32px;
    border-radius: 4px;
  ;;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders merge 3`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    
    
        background-color: #c2c2c2;
      ;;
    display: inline-flex;
    overflow: hidden;
    word-wrap: break-word;
    white-space: nowrap;
    color: #fff;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding: 4px;
  ;;
    
    width: 64px;
    height: 64px;
    font-size: 24px;
    line-height: 32px;
    border-radius: 4px;
  ;;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders size 1`] = `
Object {
  "0": "32px",
  "1": "32px",
  "2": "14px",
  "3": "22px",
}
`;

exports[`Common css renders size 2`] = `
Object {
  "0": "40px",
  "1": "40px",
  "2": "14px",
  "3": "22px",
}
`;

exports[`Common css renders size 3`] = `
Object {
  "0": "64px",
  "1": "64px",
  "2": "24px",
  "3": "32px",
}
`;

exports[`Common css renders square shape 1`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    width: 32px;
    height: 32px;
    font-size: 14px;
    line-height: 22px;
    border-radius: 4px;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders square shape 2`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    width: 40px;
    height: 40px;
    font-size: 14px;
    line-height: 22px;
    border-radius: 4px;
  ",
  "toString": [Function],
}
`;

exports[`Common css renders square shape 3`] = `
Object {
  "map": undefined,
  "next": undefined,
  "styles": "
    width: 64px;
    height: 64px;
    font-size: 24px;
    line-height: 32px;
    border-radius: 4px;
  ",
  "toString": [Function],
}
`;
